Assets/AnkrSDK/Examples/Scripts/EventListenerExample/EventListenerExample.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Numerics;
+using AnkrSDK.Core.Data;
+using AnkrSDK.Core.Data.ContractMessages.ERC721;
+using AnkrSDK.Core.Implementation;
+using AnkrSDK.Core.Infrastructure;
+using AnkrSDK.Examples.DTO;
+using AnkrSDK.Examples.ERC20Example;
+using AnkrSDK.UseCases;
+using Nethereum.RPC.Eth.DTOs;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace AnkrSDK.EventListenerExample
+{
+	public class EventListenerExample : UseCase
+	{
+		private IContract _erc20Contract;
+		private EthHandler _eth;
+		private ContractEventSubscriber _eventSubscriber;
+
+		private async void Start()
+		{
+			var ankrSDK = AnkrSDKWrapper.GetSDKInstance(ERC20ContractInformation.HttpProviderURL);
+
+			_eventSubscriber = ankrSDK.GetSubscriber(ERC20ContractInformation.WsProviderURL);
+			_eventSubscriber.OnOpenHandler += SendMint;
+			await _eventSubscriber.ListenForEvents();
```

Comment:
You should not start to listen outside SendMint() method.

Reply:
Why?

Reply:
Because we don't want to listen for events if we did not send anything

Reply:
You don't get anything because we have even separation by subscription id.

Reply:
Moreover _eventSubscriber.ListenForEvents() will block thread and next instructions will not be execute.

Reply:
If you don't want to wait for an async call you can use Forget.
UniTask is not running on Threads it is using unity playerloop.